### PR TITLE
Prevent unbounded line buffering in proxy and hook

### PR DESCRIPTION
**Pull request by ojongerius**
## What

Add memory-safe line reading functions to the MCP proxy and Claude transcript hook that enforce maximum line length limits, preventing unbounded buffer growth when reading from untrusted peers.

## Why

The standard `bufio.Reader.ReadBytes` grows its internal buffer without limit while searching for a delimiter. A malicious or malformed peer could stream an arbitrarily large line with no newline, exhausting the process's memory (issue #1010).

This change introduces bounded line readers that:
- Cap accumulated line length at a configurable maximum
- Bound memory the moment the limit is exceeded, before fully buffering the oversized line
- Maintain compatibility with normal delimited and EOF-terminated lines

## Changes

**mcp-proxy/internal/proxy/proxy.go:**
- Add `maxLineLength` constant (10 MiB) and `errLineTooLong` error
- Implement `readLine()` function that enforces the cap while reading from `bufio.Reader`
- Update `pipe()` to use `readLine()` instead of `ReadBytes()`; an oversized line closes that pipe direction (same as any other read error) rather than forwarding a truncated JSON-RPC message
- Reduce initial buffer size from 10 MB to 64 KB (no longer needed for large lines)

**hook/cmd/obsigna-hook/claude_transcript.go:**
- Add `maxTranscriptLineLength` constant (reusing daemon's `emitter.MaxFrameSize`)
- Add `errTranscriptLineTooLong` error
- Implement `readBoundedLine()` with the same bound as the proxy's `readLine()`, plus `discardRestOfLine()` to drain (without accumulating) the remainder of an abandoned oversized line so the reader resyncs at the start of the next line
- Update `lookupTranscriptUsage()` to **skip** an oversized line and keep scanning, rather than aborting the whole lookup — real transcripts can legitimately contain a large `tool_use` input (e.g. a `Write` call embedding a big file) unrelated to the id being searched for, so one oversized line must not cost every other tool call in the session its receipt enrichment. This mirrors how a malformed (non-JSON) line is already treated as a non-match rather than a fatal error.

**Tests:**
- Add `TestPipe_LineTooLong` and `TestPipe_LineAtMaxIsForwarded` for proxy line handling
- Add `TestReadLine_WithinMax` and `TestReadLine_ExceedsMax` for proxy's `readLine()` function
- Add `TestLookupTranscriptUsage_LineTooLong` (oversized line skipped, later matching line still found) and `TestLookupTranscriptUsage_OnlyOversizedLine` (file with nothing but an oversized line degrades to the ordinary not-found outcome) for the hook's transcript scanning
- Add `TestReadBoundedLine_WithinMax`, `TestReadBoundedLine_ExceedsMax`, and `TestReadBoundedLine_ExceedsMax_ResyncsAtNextLine` (forces multi-buffer-refill discard via a tiny reader buffer) for the hook's `readBoundedLine()` function

## Checklist

- [x] Tests pass for all changed components
- [x] No real keys or secrets in the diff
- [x] Input validation at trust boundaries (peer-provided data)
- [x] Edge cases tested (oversized lines, exact-limit lines, EOF-terminated lines, oversized-line resync)

### CODE REVIEW

File: hook/cmd/obsigna-hook/claude_transcript.go [hunk 1]
```diff
@@ -69,7 +88,13 @@ func lookupTranscriptUsage(path, toolUseID string) (model string, usage json.Raw
...
 	r := bufio.NewReader(f)
 	for {
-		line, readErr := r.ReadBytes('\n')
+		line, readErr := readBoundedLine(r, maxTranscriptLineLength)
+		if errors.Is(readErr, errTranscriptLineTooLong) {
+			// Can't be the JSON we're matching against — it would already
+			// have blown the daemon's own frame-size cap — so skip it like a
+			// malformed line and keep scanning instead of aborting.
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (ojongerius) [reply hook/cmd/obsigna-hook/claude_transcript.go:97]: Good catch — the description was stale. The "abort on oversized lines" behavior was the original implementation, but a code-review pass on this PR flagged that aborting the whole scan on any oversized line would silently drop enrichment for every other tool call in the transcript if a large-but-legitimate line (e.g. a `Write` tool_use embedding a big file) appeared anywhere in the file, unrelated to the id being looked up. So the code was changed to skip-and-continue (matching how a malformed line is already handled), and that's the intended, correct behavior. Updated the PR description to match.
